Sources/SwiftProtobuf/Message+BinaryAdditions.swift

```diff
@@ -97,7 +72,7 @@ extension Message {
   ///   - options: The BinaryDecodingOptions to use.
   /// - Throws: `BinaryDecodingError` if decoding fails.
   @inlinable
-  public init<Bytes: ContiguousBytes>(
```

Comment:
Can/Should `ContiguousBytes` conform to the custom protocol?  i.e. - if folks were using this api, what is their path forward?

Reply:
You cannot extend a protocol with a conformance to another protocol. I think the same applies here and we should provide this in the foundation compat module.

Reply:
I'm not sure what to do here. I've added the `Data` methods (`merge` and `init`) to the `SwiftProtobufFoundationCompat` module, which was easy enough because `Data` conforms to `SwiftProtobufContiguousBytes`. 

However in this case, as Franz said, I cannot extend `ContiguousBytes` to conform to our new protocol. As a consequence, I can't call the new methods in `SwiftProtobufCore` because objects conforming to `ContiguousBytes` may not conform to `SwiftProtobufContiguousBytes`. And I can't simply copy the implementation of the methods because `_merge` is `internal` and so is inaccessible from the FoundationCompat module (and based on the comment left on the method, it was a very conscious decision to take advantage of `@inlinable`). 
Furthermore, `_merge` uses a `BinaryDecoder` which (based on the comments too) we want to keep internal.

Any suggestions to get around this? Or should we just drop support for `ContiguousBytes`?

Reply:
We could expose `_merge` via SPI.

Reply:
I did think about this, but we support back to Swift 5.0 and I believe `@_spi` was introduced with 5.3 based on what I could find - is that correct?

Reply:
Yes, probably. We could also do the awkward workaround for older Swifts: make the type public but with a leading underscore.

Reply:
Since the protocols match, maybe skip it as all someone has to do is declare conformance for their types themselves?

Reply:
I think it will stop matching when the encoding path stops using `Data` (coming in a future PR), as we'll need to add a couple more things to it. It still shouldn't be wildly different, so I wouldn't mind just dropping it and have users conform to the new one. But if we're worried about migration then I can go for exposing the decoders but with a leading underscore, as @Lukasa suggested.

Reply:
This one is edge case enough I'm also ok dropping it completely.

@tbkka ?

Reply:
But yeah, I'm fine just dropping it for now.